src/commands/roll.ts

```diff
@@ -1,119 +1,199 @@
+            content: `**${gacha[currentGacha].rarity.toUpperCase()} JEFFREY**`,
+            embeds: [embeds[currentGacha]],
+            components: [row]
+        });
+
+        if (!gachaMessage) {
+            await interaction.editReply('Could not send message embed, please contact a developer');
             return;
         }
+
+        const buttonCollector = gachaMessage.createMessageComponentCollector({
+            filter: i => i.user.id === interaction.user.id,
+            componentType: ComponentType.Button,
+            time: TIME_IN_MS.THIRTY_MINUTES
+        });
+
+        buttonCollector.on('collect', async i => {
+
+            if (i.customId === BUTTONS.NEXT_ID) {
+                currentGacha += 1;
+                const newRow = await checkIfFirstOrLast(currentGacha, rolls);
+                await i.update({
+                    content: `**${gacha[currentGacha].rarity.toUpperCase()}**`,
+                    embeds: [embeds[currentGacha]],
+                    components: [newRow]
+                });
+            }
+
+            if (i.customId === BUTTONS.PREVIOUS_ID) {
+                currentGacha -= 1;
+                const newRow = await checkIfFirstOrLast(currentGacha, rolls);
+                await i.update({
+                    content: `**${gacha[currentGacha].rarity.toUpperCase()}**`,
+                    embeds: [embeds[currentGacha]],
+                    components: [newRow]
+                });
+            }
+
+            if (i.customId === BUTTONS.ROLL_AGAIN_ID) {
+                if (rollAgainInUse) {
+                    await i.update({})
+                    await interaction.channel?.send('Please complete your previous roll again before using this one!');
+                } else {
+                    rollAgainInUse = true;
+                    row.setComponents(BUTTONS.PREVIOUS_BUTTON, BUTTONS.NEXT_BUTTON);
+
+                    await i.update({ components: [row] });
+                    const rollAgainMsg = await interaction.channel?.send(`How many more rolls would you like to do? (Please type a number, type '0' to cancel)`);
+
+                    const msgFilter = (m: Message) => m.author.id === interaction.user.id;
+                    const msgCollector = interaction.channel!.createMessageCollector({ filter: msgFilter, time: TIME_IN_MS.TEN_MINUTES });
+
+                    msgCollector.on('collect', async m => {
+                        await tryDelete(m);
+                        if (m.content === '0') {
+                            await tryDelete(rollAgainMsg!);
+                            msgCollector.stop();
+                        }
+
+                        const content = m.content;
+                        const parsedNumber = Number(content);
+                        if (!isNaN(parsedNumber)) {
+                            msgCollector.stop();
+                            rollAgainInUse = false;
+                            await Roll.run(interaction, parsedNumber);
```

Comment:
Try not to recursively call the `run` function. You can do this in some other function, but not here.